DTO and serialization for Time Zones

RyanM-RMA opened the pull request:
Adding DTO and serialization process for time zones, so we can use the newer java json/xml serialization/deserialization

Fixes #499

RyanM-RMA committed:
Resolving array indexing issue

RyanM-RMA committed:
Adding DTO and aliasing for time zones

rma-rripken commented:
Can we get  a unit test of just the TimeZone DTO serialization by itself?   Include a json or xml example file.

RyanM-RMA committed:
Adding unit tests for TimeZone validation

Removing unused import from time zone

rma-rripken commented:
I'd like to see a resource file with an example of the serialized format.

RyanM-RMA commented:
oh, actually I should also get a test for the List<TimeZone> in there too, since that's what we're serializing with getAll.  I totally meant to do that.

RyanM-RMA committed:
Adding json resource, and TimeZones class

Added TimeZones class so we can deserialize to and from a list object like LocationLevels.
